scrape/target.go

```diff
@@ -492,3 +503,53 @@ func TargetsFromGroup(tg *targetgroup.Group, cfg *config.ScrapeConfig) ([]*Targe
 	}
 	return targets, failures
 }
+
+// buildTargetLabels builds final target labels based on the given target label set, target group label set and scrape config.
+func buildTargetLabels(targetLSet, targetGroupLSet model.LabelSet, cfg *config.ScrapeConfig) labels.Labels {
+	labelsMap := make(map[string]string, len(targetLSet)+len(targetGroupLSet))
+	// Labels from scrape config.
+	scrapeLabels := []labels.Label{
+		{Name: model.JobLabel, Value: cfg.JobName},
+		{Name: model.ScrapeIntervalLabel, Value: cfg.ScrapeInterval.String()},
+		{Name: model.ScrapeTimeoutLabel, Value: cfg.ScrapeTimeout.String()},
+		{Name: model.MetricsPathLabel, Value: cfg.MetricsPath},
+		{Name: model.SchemeLabel, Value: cfg.Scheme},
+	}
+	for _, l := range scrapeLabels {
+		set(labelsMap, l.Name, l.Value)
+	}
+	// Target group labels.
+	for ln, lv := range targetGroupLSet {
+		set(labelsMap, string(ln), string(lv))
+	}
+
+	// Target labels.
+	for ln, lv := range targetLSet {
+		set(labelsMap, string(ln), string(lv))
+	}
+
+	// Labels from config query params.
+	for k, v := range cfg.Params {
```

Comment:
it does not sound right that this comes last.

Reply:
In this context, I don't think the order matter. It won't change the final result in my opinion.